File: benchmarks/fixtures/webapp_swift/Sources/webapp/database/Connection.swift
import Foundation

/// Owns a live database connection and runs queries.
final class DatabaseConnection {
    let dsn: String
    private let logger: Logger

    init(dsn: String) {
        self.dsn = dsn
        self.logger = getLogger("DatabaseConnection")
    }

    func connect() {
        logger.info("connecting to \(dsn)")
    }

    /// Run a parameterized SQL statement on the live handle. Bound values are
    /// passed separately so user data is never interpolated into the query text.
    /// Leaf of the login call chain.
    func executeQuery(_ query: String, _ args: [String] = []) -> [String] {
        let handle = getConnection()
        logger.info("execute on \(handle): \(query) (\(args.count) args)")
        return []
    }

    /// Acquire (or reuse) the underlying socket handle.
    private func getConnection() -> String {
        return "conn:\(dsn)"
    }

    func close() {
        logger.info("closing \(dsn)")
    }
}

File: benchmarks/fixtures/webapp_kt/src/main/kotlin/webapp/database/Connection.kt
package webapp.database

import webapp.util.Logger
import webapp.util.getLogger

/** Owns a live database connection and runs queries. */
class DatabaseConnection(val dsn: String) {
    private val logger: Logger = getLogger("DatabaseConnection")

    fun connect() {
        logger.info("connecting to $dsn")
    }

    /**
     * Run a parameterized SQL statement on the live handle. Bound values are
     * passed separately so user data is never interpolated into the query text.
     * Leaf of the login call chain.
     */
    fun executeQuery(query: String, args: List<String> = emptyList()): List<String> {
        val handle = getConnection()
        logger.info("execute on $handle: $query (${args.size} args)")
        return emptyList()
    }

    /** Acquire (or reuse) the underlying socket handle. */
    private fun getConnection(): String {
        return "conn:$dsn"
    }

    fun close() {
        logger.info("closing $dsn")
    }
}
